setup.py

```diff
@@ -0,0 +1,48 @@
+import os
+from setuptools import setup
+
+from undate import __version__
+
+# load the readme contents and use for package long description
+with open(os.path.join(os.path.dirname(__file__), "README.md")) as readme:
+    README = readme.read()
+
+REQUIREMENTS = []
+TEST_REQUIREMENTS = ["pytest>=7.2"]
+DEV_REQUIREMENTS = [
+    "black>=22.10.0",
+    "pre-commit>=2.20.0",
+]
+
+setup(
+    name="undate",
+    version=__version__,
+    description="library for working with uncertain, fuzzy, or "
+    + "partially unknown dates and date intervals",
+    long_description=README,
+    url="https://github.com/dh-tech/hackathon-2022",
+    author="DHTech",
+    # author_email='', ?
```

Comment:
do we have a dh-tech email?

Reply:
no, but maybe we should create one? Maybe a gmail one that we can share?

Reply:
seems like a good idea

Reply:
Any reason to use `setup.py` vs `setup.cfg` + `pyproject.toml`? Think I've usually followed a structure somewhat like this for packages I think will be distributed: https://setuptools.pypa.io/en/latest/userguide/declarative_config.html.

Reply:
Because I am old and this is what I've always used ? 👵 😅 

I'm open to this approach, just haven't used it before. What are the advantages? Do you want to revise it to use this format?  

(Does it have to be toml or are other formats like yaml supported too?)

Reply:
No complaints if this works! I haven't created that many public packages, so this is just what I know of. I think it's because `setup.py` is usually declarative anyway, so why not move that to an explicitly declarative file. Then `setuptools` can just parse that directly for the metadata. I think it has to be .toml. `pyproject.toml` makes Pypi distribution easier and you can set linting options there - this is an example from a different project.

### pyproject.toml
```
[build-system]
requires = ["setuptools>=42"]
build-backend = "setuptools.build_meta"

[tool.black]
# will not convert single-quote strings to double-quote
skip-string-normalization = true

[tool.isort]
# play nice with black formatter
# see https://pycqa.github.io/isort/docs/configuration/black_compatibility.html
profile = "black"

[tool.pytest.ini_options]
minversion = "7.0"
testpaths = ["tests"]
pythonpath = "src"
```

### setup.cfg
```
[metadata]
name = IIIFingest
version = 1.1.1
author = Cole Crawford
author_email = cole_crawford@fas.harvard.edu
description = A wrapper library for interacting with the MPS ingest system for IIIF resources
long_description = file: readme.md
long_description_content_type = text/markdown
url = https://github.com/Harvard-ATG/lts-iiif-ingest-service
project_urls =
    Bug Tracker = https://github.com/Harvard-ATG/lts-iiif-ingest-service/issues
classifiers =
    Programming Language :: Python :: 3
    Operating System :: OS Independent
    License :: OSI Approved :: Apache Software License

[options]
package_dir =
    = src
packages = find:
python_requires = >=3.7
install_requires = 
    pyjwt
    cryptography
    jsonschema
    boto3
    Pillow
    requests
    shortuuid
    pyIIIFpres

[options.packages.find]
where = src
```

Then setup.py becomes simple:
```
import setuptools
if __name__ == "__main__":
    setuptools.setup()
```

And PyPi distribution is pretty straightforward:
```
$ python setup.py sdist bdist_wheel
$ tar tzf dist/IIIFingest-{VERSION}.tar.gz
$ twine upload dist/IIIFingest-{VERSION}*
```
Haven't used Github for Python package distribution, I think for PHP I've bundled and then uploaded manually which was a little clunky.

Reply:
It looks like this format is the new recommended approach ([PEP-0518](https://peps.python.org/pep-0518/)) make it installable with more tools.  I can take a stab at refactoring to that later on today.

Reply:
Actually, let's go ahead and merge this and then I'll make an issue refactor to the new recomended format as a second step.